app/views/match_contacts/_match_contacts_list.haml

```diff
@@ -9,12 +9,17 @@
       - Translation.translate('Stabilization Service Providers')
       - Translation.translate('Housing Search Providers')
       .c-detail-group__title--secondary= Translation.translate(title)
-      .detail-box--value
-        - match.send(contact_type).each do |contact|
+      .detail-box--value 
+        - contact_type_column = contact_type.to_s.sub('_contacts','').to_sym #TODO: This feels too hacky
+        - all_contacts = match.client_opportunity_match_contacts.where(ClientOpportunityMatchContact.arel_table[contact_type_column.to_sym].eq(true)).index_by(&:contact_id)
```

Comment:
Alternatively (or maybe in addition), there could be a scope on `ClientOpportunityMatchContact` that takes the `contact_type` and returns where the column is true.  This could then be:
```suggestion
        - all_contacts = match.client_opportunity_match_contacts.for_contact_type(contact_type).index_by(&:contact_id)
```